### PR TITLE
Minimal frontend

### PR DESCRIPTION
**Pull request by SebastiaanKloos**
Some users like to install their own version of LaraBug. This PR should make the app feel a bit more of an app as it disables some unnecessary pages / links.
CleanShot 2021-05-05 at 00 14 36@2x

CleanShot 2021-05-05 at 00 11

### CODE REVIEW

**Comment by Cannonb4ll** [comment]
Can you change all the `env()` calls to `config()` and move that to larabug.php config file?

This because of config caching.

File: app/Http/Middleware/MinimalFrontendMode.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  string|null  $guard
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $guard = null)
+    {
+        if (!Auth::guard($guard)->check() && env('MINIMAL_FRONTEND')) {
```
Comment (Glennmen) [inline app/Http/Middleware/MinimalFrontendMode.php:20]: What does this exactly do? 🤔 
If you aren't logged in and `MINIMAL_FRONTEND` is true you are redirected to the login page, but if you aren't logged in and `MINIMAL_FRONTEND` is false you are not redirected. I am confused why this is needed and why the normal `auth` flow can't be used.

Comment (SebastiaanKloos) [reply app/Http/Middleware/MinimalFrontendMode.php:20]: @Glennmen I've added this to disable the frontend on self hosted versions. You don't need the marketing pages as it's your own installation (might be wrong for SEO too)

Comment (Glennmen) [reply app/Http/Middleware/MinimalFrontendMode.php:20]: Thanks, I see it now

**Review by Glennmen — APPROVED** [review]
Besides Dennis his comment all the rest looks fine.

**Comment by SebastiaanKloos** [comment]
@Cannonb4ll Are you okay with the default value being true?  Only the hosted version will need the .env variable, so the frontend would be enabled

`MINIMAL_FRONTEND=false`

File: config/larabug.php [hunk 2]
```diff
@@ -38,4 +38,9 @@
         'Symfony\Component\HttpKernel\Exception\NotFoundHttpException',
     ],
 
+    /**
+     * Use minimal frontend mode
+     */
+    'minimal_frontend' => env('MINIMAL_FRONTEND', true),
```
Comment (Glennmen) [inline config/larabug.php:44]: Shouldn't the default be `false`?

Comment (Glennmen) [reply config/larabug.php:44]: Also as a best practice I would add this new environment variable to `.env.example` with no value.

**Review by Glennmen — CHANGES_REQUESTED** [review]

**Comment by Glennmen** [comment]
Oh didn't see you comment till after I reviewed the code.

**Comment by Cannonb4ll** [comment]
Yes default to false is fine with me, I'll force to true in our case as we want to have that available of course.

**Comment by SebastiaanKloos** [comment]
> Yes default to false is fine with me, I'll force to true in our case as we want to have that available of course.

@Cannonb4ll you should set it to false in your case

**Pull request merged**